app/lib/shared/integrity.dart

```diff
@@ -62,17 +62,20 @@ class IntegrityChecker {
   /// Runs integrity checks, and returns the list of problems.
   Stream<String> findProblems() async* {
     _httpClient = httpRetryClient();
```

Comment:
Any concerns that this could be re-entrant?

Not sure how to "lock" that – just a though.

Reply:
rewrite it to a function instead of misusing a class.